### ISSUE TITLE
LazyAdam

### ISSUE THREAD

**Issue by bitfort** [mlcommons/training_policies#78]
We believe there are cases where LazyAdam is necessary for certain models to be scaled up. It is worth discussing the option of opening up Optimizer choices under certain conditions.

**Comment by bitfort** [comment]
SWG:

Across any use of Adam or LazyAdam for v0.5 we tentatively agree that the two can be swapped at the choice of the submitter. Confirmation to follow in next meeting.

**Comment by bitfort** [comment]
SWG:

Will revisit in a future meeting or offline on the submitters mailing list.

**Comment by bitfort** [comment]
Context:
We talk about LazyAdam v. Adam and setting the betas. Longer term, we want rules for Adam (similar to our LazyAdam rules on scaling learning rate). 

SWG:
Tentatively:
- You can swap Adam and LazyAdam at the submitter's discretion for any benchmark. If you swap, you may re-tune the parameters at your discretion. Intention to swap should be announced to the submitters group by October 15th, present data to demonstrate the swap is necessary, in addition to data to support tuning of any particular parameters.

**Comment by bitfort** [comment]
SWG:

We formalize our tentative decision as:

For NCF, either Adam or LazyAdam may be used; Adam parameters may be tuned freely (step and betas). For transformer, Adam or LazyAdam may be used but the betas cannot be further tuned without additional data. 

Use of Adam and LazyAdam are subject to sharing as are other hyperparameters.

**Comment by nvpaulius** [comment]
I'd like to clarify whether by `step` we mean `alpha` using the notation in Algorithm 1 of the original Adam paper (https://arxiv.org/pdf/1412.6980.pdf). Is that the case?
It may be good to formalize the names, since framework implementations tend to call this `lr`.

Also, we should clarify that by "... may be tuned freely ..." we mean that values can be chosen freely, but they must stay fixed throughout training (conforming to what the NCF reference does).
